src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Handlers/CommandLineItemHandler.cs

```diff
@@ -53,9 +53,11 @@ public override Task HandleAsync(IProjectVersionedValue<IProjectSubscriptionUpda
             if (!ProcessDesignTimeBuildFailure(projectChange, context))
             {
                 ProcessOptions(projectChange, context);
-                ProcessItems(projectChange, context, isActiveContext);
             }
 
+            // We need to process items even from failed design time builds, as we might be removing some items.
```

Comment:
@davkean I attached a debugger and see that when I change TF, and stop at the first `IProjectSubscriptionUpdate` with only removed items, we have a failed design time build in the debug output window, where the build is attempting to find reference assemblies for the previous target framework. For example, after switching from netstandard1.3 to net451, I see the following error in the debug output window:

```
Target "GetReferenceAssemblyPaths" in file "C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\Bin\Microsoft.Common.CurrentVersion.targets" from project "C:\Users\mavasani\Documents\Visual Studio 2017\Projects\ConsoleApp12\ConsoleApp12\ConsoleApp12.csproj" (target "PrepareForBuild" depends on it):
The reference assemblies for framework ".NETStandard,Version=v1.4" were not found. To resolve this, install the SDK or Targeting Pack for this framework version or retarget your application to a version of the framework for which you have the SDK or Targeting Pack installed. Note that assemblies will be resolved from the Global Assembly Cache (GAC) and will be used in place of reference assemblies. Therefore your assembly may not be correctly targeted for the framework you intend.
Done building target "GetReferenceAssemblyPaths" in project "ConsoleApp12.csproj" -- FAILED.
Done building project "ConsoleApp12.csproj" -- FAILED.
Build FAILED.
```

After a while, I get a second update with added items. I am pretty sure that this failed design time build is responsible for the underlying bug.